systems/notification/mailer/pkg/server/mailer.go

```diff
@@ -22,139 +21,165 @@ import (
 	"github.com/ukama/ukama/systems/notification/mailer/pkg/db"
 )
 
-type EmailData struct {
-	To      []string
-	Subject string
-	Body    string
-	Values  map[string]interface{}
+type EmailPayload struct {
+	To           []string
+	TemplateName string `json:"template_name"`
+	Values       map[string]interface{}
 }
 
 type MailerServer struct {
-	mailerRepoRepo db.MailerRepo
+	mailerRepo db.MailerRepo
 	pb.UnimplementedMailerServiceServer
-	mailer *pkg.Mailer
+	mailer        *pkg.Mailer
+	templatesPath string
 }
 
-func NewMailerServer(mailerRepoRepo db.MailerRepo, mail *pkg.Mailer) *MailerServer {
+func NewMailerServer(mailerRepo db.MailerRepo, mail *pkg.Mailer, templatesPath string) *MailerServer {
 	return &MailerServer{
-		mailerRepoRepo: mailerRepoRepo,
-		mailer:         mail,
+		mailerRepo:    mailerRepo,
+		mailer:        mail,
+		templatesPath: templatesPath,
 	}
 }
 
 func (s *MailerServer) SendEmail(ctx context.Context, req *pb.SendEmailRequest) (*pb.SendEmailResponse, error) {
-
-	if req.To == nil || req.Subject == "" || req.Body == "" {
-		return nil, errors.New("missing required fields in SendEmailRequest")
-	}
+	log.Infof("Sending email to %v", req.To)
+	values := make(map[string]interface{})
 	mailId := uuid.NewV4()
-	currentTime := time.Now()
-	sentAt := &currentTime
-
-	from := s.mailer.From
-	to := req.GetTo()
-	subject := req.GetSubject()
-	bodyTemplate := req.GetBody()
-	values := req.GetValues()
-	emailData := &EmailData{
-		Subject: subject,
-		Body:    bodyTemplate,
-		Values:  make(map[string]interface{}),
+	for key, value := range req.Values {
+		values[key] = value
 	}
 
-	for key, value := range values {
-		emailData.Values[key] = value
+	payload := &EmailPayload{
+		To:           req.To,
+		TemplateName: req.TemplateName,
+		Values:       values,
 	}
-	emailData.Values["EmailID"] = mailId.String()
-	tmpl, err := template.New("email").Parse(emailData.Body)
+
+	body, err := s.prepareMsg(payload)
 	if err != nil {
-		log.Errorf("Failed to parse email template: %v", err)
-		return nil, err
+		return nil, status.Error(codes.Internal, "failed to prepare email body")
 	}
 
-	var bodyBuffer bytes.Buffer
-	err = tmpl.Execute(&bodyBuffer, emailData.Values)
+	c, err := smtp.Dial(fmt.Sprintf("%s:%d", s.mailer.Host, s.mailer.Port))
 	if err != nil {
-		log.Errorf("Failed to render email template: %v", err)
-		return nil, err
+		return nil, status.Error(codes.Internal, "failed to connect to SMTP server")
```

Comment:
always user `status.Errorf ` to include the original err in the description.

Reply:
Addressed.